src/index.ts

```diff
@@ -200,38 +192,41 @@ export const build = <FactoryResultType>(
   };
 };
 
-export const oneOf = <T>(...options: T[]): OneOfGenerator => {
+export const oneOf = <T>(...options: T[]): OneOfGenerator<T> => {
   return {
     generatorType: 'oneOf',
-    options,
-    call: <T>(options: T[]): T => {
+    call: (): T => {
       const randomIndex = Math.floor(Math.random() * options.length);
 
       return options[randomIndex];
     },
   };
 };
 
-export const bool = (): OneOfGenerator => oneOf(true, false);
+export const bool = () => oneOf(true, false);
+
+type Sequence = {
+  (): SequenceGenerator<number>;
+  <T>(userProvidedFunction: (count: number) => T): SequenceGenerator<T>;
+};
 
-export const sequence = (
-  userProvidedFunction: SequenceFunction = (x) => x
-): SequenceGenerator => {
+export const sequence = ((userProvidedFunction) => {
   return {
     generatorType: 'sequence',
-    userProvidedFunction,
-    call: (userProvidedFunction: SequenceFunction, counter: number) => {
+    call: (counter: number) => {
+      if (typeof userProvidedFunction === 'undefined') {
+        return counter;
+      }
       return userProvidedFunction(counter);
     },
   };
-};
+}) as Sequence;
```

Comment:
Please see [this stackoverflow answer](https://stackoverflow.com/a/61366790) as to why we need to do this for `sequence`.